builder/image/Dockerfile

```diff
@@ -13,28 +11,32 @@ RUN curl -sSL -o /usr/local/bin/etcdctl https://s3-us-west-2.amazonaws.com/opdem
 RUN curl -sSL -o /usr/local/bin/confd https://github.com/kelseyhightower/confd/releases/download/v0.9.0/confd-0.9.0-linux-amd64 \
     && chmod +x /usr/local/bin/confd
 
-# install docker-in-docker
-RUN echo "deb http://get.docker.com/ubuntu docker main" > /etc/apt/sources.list.d/docker.list
-RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 36A1D7869245C8950F966E92D8576A8BA88D21E9
+RUN apk add --update-cache \
+    coreutils \
+    device-mapper \
+    e2fsprogs \
+    git \
+    iptables \
+    libudev \
+    lxc \
+    openssh \
+    udev \
+    util-linux \
+    xz \
+    && rm -rf /var/cache/apk/*
 
-# install builder, docker, and hook dependencies
-RUN apt-get update && apt-get install -yq \
-    openssh-server git \
-    aufs-tools iptables lxc \
-    lxc-docker-1.5.0
+# the docker package in alpine disables aufs and devicemapper
+RUN curl -sSL https://get.docker.com/builds/Linux/x86_64/docker-1.5.0 -o /usr/bin/docker && \
+  chmod +x /usr/bin/docker
 
 # configure ssh server
-RUN rm /etc/ssh/ssh_host_*
-RUN dpkg-reconfigure openssh-server
-RUN mkdir -p /var/run/sshd
-
-# configure locale
-RUN echo LANG="en_US.UTF-8" > /etc/default/locale && dpkg-reconfigure locales
+RUN mkdir -p /var/run/sshd && rm -rf /etc/ssh/ssh_host*
```

Comment:
Should we move this to bin/boot in a separate commit so we can also close #3132?